distsql: pre-canned values processor

RaduBerinde opened the pull request:
Adding a processor that just spits out a set of pre-canned values. Useful when
queries involve VALUES.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13126)
<!-- Reviewable:end -->

RaduBerinde committed:
distsql: pre-canned values processor

Adding a processor that just spits out a set of pre-canned values. Useful when
queries involve VALUES.

andreimatei commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/distsqlrun/processors.proto, line 106 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13126#-KbQdyWtkh2RrkG56bJy:-KbQdyWtkh2RrkG56bJz:b-fvtj4v) ([raw file](https://github.com/cockroachdb/cockroach/blob/c083c2593cfe3449ad44f6dcc574eeb0cf7bd680/pkg/sql/distsqlrun/processors.proto#L106)):*
> ```protobuf
> message ValuesCoreSpec {
>   // There is one DatumInfo for each element in a row.
>   repeated DatumInfo info = 1 [(gogoproto.nullable) = false];
> ```

`columns`?

---

*[pkg/sql/distsqlrun/values.go, line 26 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13126#-KbQeJS2bt1Ffq0LMnzl:-KbQeJS2bt1Ffq0LMnzm:bw6ip8y) ([raw file](https://github.com/cockroachdb/cockroach/blob/c083c2593cfe3449ad44f6dcc574eeb0cf7bd680/pkg/sql/distsqlrun/values.go#L26)):*
> ```Go
> )
> 
> // valuesProcessor is a processor that simply passes rows through from the
> ```

is this comment accurate? It does pass rows from any synchronizer, does it? That's the noop proc.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13126#-:-KbQeaBLfynIhJ2Lf1_g:bnfp4nl)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/distsqlrun/processors.proto, line 106 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13126#-KbQdyWtkh2RrkG56bJy:-KbRBU4-5xompdX-sUQh:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c083c2593cfe3449ad44f6dcc574eeb0cf7bd680/pkg/sql/distsqlrun/processors.proto#L106)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

`columns`?
</blockquote></details>

Done.

---

*[pkg/sql/distsqlrun/values.go, line 26 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13126#-KbQeJS2bt1Ffq0LMnzl:-KbRBUJbTSufYC0StXsh:byi28b7) ([raw file](https://github.com/cockroachdb/cockroach/blob/c083c2593cfe3449ad44f6dcc574eeb0cf7bd680/pkg/sql/distsqlrun/values.go#L26)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

is this comment accurate? It does pass rows from any synchronizer, does it? That's the noop proc.
</blockquote></details>

Fixed, as a copy/paste-o. I updated the comments for noop while i'm at it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13126)*
<!-- Sent from Reviewable.io -->
